pkg/activator/net/throttler.go

```diff
@@ -541,6 +566,40 @@ func (t *Throttler) handleUpdate(update revisionDestsUpdate) {
 	}
 }
 
+func (t *Throttler) handlePubEpsUpdate(eps *corev1.Endpoints) {
+	t.logger.Infof("Public EPS updates: %#v", eps)
+
+	revN := eps.Labels[serving.RevisionLabelKey]
+	if revN == "" {
+		// Perhops, we're not the only ones using the same selector label.
+		t.logger.Infof("Ignoring update for PublicService %s/%s", eps.Namespace, eps.Name)
+		return
+	}
+	rev := types.NamespacedName{Name: revN, Namespace: eps.Namespace}
+	if rt, err := t.getOrCreateRevisionThrottler(rev); err != nil {
+		logger := t.logger.With(zap.Any(logkey.Key, rev))
+		if k8serrors.IsNotFound(err) {
+			logger.Debug("Revision not found. It was probably removed")
+		} else {
+			logger.Errorw("Failed to get revision throttler", zap.Error(err))
+		}
+	} else {
+		rt.handlePubEpsUpdate(eps, t.ipAddress)
+	}
+
+}
+
+func (rt *revisionThrottler) handlePubEpsUpdate(eps *corev1.Endpoints, selfIP string) {
+	// NB: this is guaranteed to be executed on a single thread.
+	epSet, _ := endpointsToDests(eps, networking.ServicePortNameHTTP1)
```

Comment:
As you mentioned in Slack, this will need to get the servicename based on the passed in proto.

Reply:
yep, correct (hence wip) 😊

Reply:
Done. This required some tweaking of how we pass the activator self IP, but otherwise was painless.